tests/test_icon_score_engine.py

```diff
@@ -17,130 +17,217 @@
-
-        self.engine = IconScoreEngine()
-        # Use mock to prevent an exception from IconScoreEngine._validate_score_blacklist().
-        IconScoreEngine._validate_score_blacklist = mock.Mock()
-
-        self._from = create_address(AddressPrefix.EOA)
-        self._icon_score_address = create_address(AddressPrefix.CONTRACT)
-
-        IconScoreContext.icon_score_deploy_engine = deploy_engine
-        self._context = IconScoreContext(IconScoreContextType.DIRECT)
-        self._context.msg = Message(self._from, 0)
-        tx_hash = create_tx_hash()
-        self._context.tx = Transaction(
-            tx_hash, origin=create_address(AddressPrefix.EOA))
-        block_hash = create_block_hash()
-        self._context.block = Block(1, block_hash, 0, None)
-
-    def tearDown(self):
-        self.engine = None
-        self.icx_storage.close(self._context)
-        ContextDatabaseFactory.close()
-
-        path = os.path.join(TEST_ROOT_PATH, self._SCORE_ROOT_PATH)
-        remove_path(path)
-        path = os.path.join(TEST_ROOT_PATH, self._TEST_DB_PATH)
-        remove_path(path)
-
-        IconScoreClassLoader.exit(self._SCORE_ROOT_PATH)
-
-    @staticmethod
-    def read_zipfile_as_byte(archive_path: str) -> bytes:
-        with open(archive_path, 'rb') as f:
-            byte_data = f.read()
-            return byte_data
-
-    @staticmethod
-    def __unpack_zip_file(install_path: str, data: bytes):
-        file_info_generator = IconScoreDeployer._extract_files_gen(data)
-        for name, file_info, parent_directory in file_info_generator:
-            if not os.path.exists(os.path.join(install_path, parent_directory)):
-                os.makedirs(os.path.join(install_path, parent_directory))
-            with file_info as file_info_context,\
-                    open(os.path.join(install_path, name), 'wb') as dest:
-                contents = file_info_context.read()
-                dest.write(contents)
-        return True
-
-    @staticmethod
-    def __ensure_dir(file_path):
-        directory = os.path.dirname(file_path)
-        if not os.path.exists(directory):
-            os.makedirs(directory)
-
-    def test_call_method(self):
-        calldata = {'method': 'total_supply', 'params': {}}
-
-        # proj_name = 'test_score'
-        # path = os.path.join(TEST_ROOT_PATH, 'tests/tmp/{}'.format(proj_name))
-        # install_data = {'contentType': 'application/tbears', 'content': path}
-        # self._engine.invoke(
-        #     self._context, self._icon_score_address, 'install', install_data)
-        # self._engine.commit(self._context)
+    @patch('iconservice.iconscore.icon_score_context_util.IconScoreContextUtil.validate_score_blacklist')
+    def test_validate_score_blacklist(self,
+                                      mocked_score_context_util_validate_score_blacklist):
+        context = IconScoreContext(IconScoreContextType.INVOKE)
+
+        # failure case: should not accept ZERO_SCORE_ADDRESS as SCORE address
```

Comment:
In my opinion,
we need to divide function by case if it is possible.
we should determine about it through this and #47